include/deal.II/base/point.h

```diff
@@ -176,33 +188,47 @@ class Point : public Tensor<1, dim, Number>
    *
    * @note This function can also be used in CUDA device code.
    */
-  static DEAL_II_CUDA_HOST_DEV Point<dim, Number>
-                               unit_vector(const unsigned int i);
+  static DEAL_II_CUDA_HOST_DEV DEAL_II_CONSTEXPR Point<dim, Number>
+                                                 unit_vector(const unsigned int i);
 
   /**
    * Read access to the <tt>index</tt>th coordinate.
    *
    * @note This function can also be used in CUDA device code.
    */
-  DEAL_II_CUDA_HOST_DEV Number
-                        operator()(const unsigned int index) const;
+  DEAL_II_CUDA_HOST_DEV DEAL_II_CONSTEXPR Number
+                                          operator()(const unsigned int index) const;
 
   /**
    * Read and write access to the <tt>index</tt>th coordinate.
    *
    * @note This function can also be used in CUDA device code.
    */
-  DEAL_II_CUDA_HOST_DEV Number &
-                        operator()(const unsigned int index);
+  DEAL_II_CUDA_HOST_DEV DEAL_II_CONSTEXPR Number &
+                                          operator()(const unsigned int index);
 
   /**
    * Assignment operator from Tensor<1, dim, Number> with different underlying
    * scalar type. This obviously requires that the @p OtherNumber type is
    * convertible to @p Number.
    */
   template <typename OtherNumber>
-  Point<dim, Number> &
-  operator=(const Tensor<1, dim, OtherNumber> &p);
+  DEAL_II_CONSTEXPR Point<dim, Number> &
+                    operator=(const Tensor<1, dim, OtherNumber> &p);
+
+  /**
+   * Converts the Point<dim> to a Point<dim + 1> object by appending a
+   * coordinate value as the last coordinate
+   */
+  DEAL_II_CONSTEXPR DEAL_II_CUDA_HOST_DEV Point<dim + 1, Number>
+                                          extend(const Number extension) const;
```

Comment:
Adding this function allows the simplification I made in the file [data_out_stack.cc](https://github.com/dealii/dealii/pull/8673/files#diff-943de848b956aaea862c5d1f5a09bb00)

Within `data_out_stack.cc` we need a non-zero value for the argument `extension`.

The changes and the introduction of the `extend` method are necessary to prevent compile error in `data_out_stack.cc` due to `static_assert` in the constructors of `Point`. Basically, whenever there is a `switch (dim)` statement and it constructs `Point<dim>` within that block, we need to modify it to allow `static_assert` in constructors.

Reply:
I see.

As I mentioned before, I think the addition of the `extend`/`embed` function by itself, and its use in `data_out_stack.cc` makes sense, and could be a patch of itself.